src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,140 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> prev;
+        private Node<T> next;
+
+        private Node(Node<T> prev, T element, Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(last, value, null);
+        if (size == 0) {
+            first = newNode;
+        } else {
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdd(index);
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> node = getNode(index);
+        Node<T> newNode = new Node<>(node.prev, value, node);
+        if (index == 0) {
+            first = newNode;
+        } else {
+            node.prev.next = newNode;
+        }
+        node.prev = newNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T item : list) {
+            add(item);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return getNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        T oldValue = getNode(index).item;
+        getNode(index).item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> node = getNode(index);
+        return unlink(node);
     }
 
     @Override
     public boolean remove(T object) {
+        for (Node<T> i = first; i != null; i = i.next) {
+            if ((object != null && object.equals(i.item)) || i.item == object) {
+                unlink(i);
+                return true;
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void checkIndexForAdd(int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index does not exist: " + index);
+        }
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index does not exist: " + index);
+        }
+    }
+
+    private Node<T> getNode(int index) {
+        checkIndex(index);
+        if (index > (size >> 1)) {
+            Node<T> tempNode = first;
```

Comment:
No need to declare two different nodes in both if and else methods. You can declare it once after checkIndex.
Also maybe it will be better to write size / 2 in if condition?